examples/timeseries_peak_finding.py

```diff
@@ -0,0 +1,118 @@
+"""
+==============================
+Find Peaks in SunPy TimeSeries
+==============================
+
+This example illustrates how to find minimum or maximum peaks in a TimeSeries.
+Note: Peak finding is a complex problem that has many potential solutions and
+this example is just one method of many.
+"""
+
+##############################################################################
+# Start by importing the necessary modules.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+import sunpy.data.sample
+from sunpy.timeseries import TimeSeries
+
+##############################################################################
+# We will now create a TimeSeries object from an observational data source,
+# Also, we will truncate it to do analysis on a smaller time duration of 10
+# years.
+
+ts_noaa_ind = sunpy.timeseries.TimeSeries(
+    sunpy.data.sample.NOAAINDICES_TIMESERIES, source='NOAAIndices')
+my_timeseries = ts_noaa_ind.truncate('1991/01/01', '2001/01/01')
+my_timeseries.peek()
+
+##############################################################################
+# To find extrema in any TimeSeries, we first define a function findpeaks that
+# takes in input an iterable data series and a DELTA value. The DELTA value
+# controls how much difference between values in the TimeSeries defines an
+# extremum point. Inside the function, we iterate over the data values of
+# TimeSeries and consider a point to be a local maxima if it has the maximal
+# value, and was preceded (to the left) by a value lower by DELTA. Similar
+# logic applies to find a local minima.
+
+
+def findpeaks(series, DELTA):
+    """
+    Finds extrema in a pandas series data.
+
+    Parameters
+    ----------
+    series : `pandas.Series`
```

Comment:
Is a pandas.Series or a Timeseries? or both?

Reply:
Using `iteritems()` on Timeseries results in
``` AttributeError: 'GenericTimeSeries' object has no attribute 'iteritems'```
Moreover, it doesn't make sense of asking to find peaks in Timeseries which contains many columns. Would it return [Min, Max] lists for every column?